src/components/CharactersPage/CharactersTable/CharactersTable.js

```diff
@@ -86,11 +92,66 @@ const CharactersTable = ({ onRowClick }) => {
     },
   ]
 
+  const rows = useMemo(() => {
+    return data.allPeople.edges.map((edge) => edge.node)
+  }, [data])
+
+  const pageInfo = data.allPeople.pageInfo
+  const totalRowsCount = data.allPeople.totalCount
```

Comment:
Move this to the parent component and pass down the prepared information this component needs. Provide sensible defaults as needed